compressors/elias_delta_uint_coder.py

```diff
@@ -0,0 +1,164 @@
+"""Elias Delta Universal uint encoder
+
+https://en.wikipedia.org/wiki/Elias_delta_coding
+
+Idea:
+To encode X >=0 :
+1. Let Y = X+1 so Y > 0
+2. Let N = floor(log_2 (Y)) [this is 1 less than the number of bits in binary repr of Y]. Note N>=0
+3. Let M = N+1 so M > 0
+4. Let L = floor(log_2 (M)) [this is 1 less than the number of bits in binary repr of M].
+5. Write L zeros, followed by
+6. The L+1 bit binary representation of M(=N+1) [note this always begins with a 1], followed by
+7. All but the leading bit of Y(=X+1) (N bits) [leading bit is always 1 so we skip]
+
+Decoding is quite simple:
+- first we read the value of L by finding the number of 0s before a 1
+- then we read L+1 bits to compute M
```

Comment:
why did we have to use Elias delta coder? is the Golomb code inefficient or has different probabilities?

Reply:
Golomb is more suited for geometric distribution where the probabilities fall really fast. Here we often get larger values and wanted something that uses around log n bits to encode n, so more of a 1/n type probability decay. I looked the gzip and zstd codes and they both were using this sort of decay.